### PR TITLE
[Repec] upgrade ruby

### PR DESCRIPTION
**Pull request by sandbergja**
Also refactor playbook/group_vars to use the `runtime_env` and `common.yml` patterns.

Sibling PR: https://github.com/pulibrary/repecwp/pull/81

### CODE REVIEW

File: playbooks/repec.yml [hunk 1]
```diff
@@ -1,9 +1,12 @@
...
+# by default this playbook runs in the staging environment
+# to run in production, pass '-e runtime_env=production'
+- hosts: repec_{{ runtime_env | default('staging') }}
   remote_user: pulsys
   become: true
   vars_files:
-    - ../group_vars/repec/repec_prod.yml
+    - ../group_vars/repec/repec_{{ runtime_env | default('staging') }}.yml
+    - ../group_vars/repec/common.yml
     - ../group_vars/repec/vault.yml
```
Comment (maxkadel) [inline playbooks/repec.yml:10]: Should this playbook include the datadog role for production? Is it a conscious choice to leave it off?

Comment (maxkadel) [inline playbooks/repec.yml:10]: Is there something that is supposed to happen with the assets repository or something after you run this playbook? Would that be good to add to the post_tasks?

**Review by maxkadel — APPROVED** [review]
Thanks for the upgrade and refactor!

Approved, but just double check me on the comments.

File: playbooks/repec.yml (hunk 1 above)
Comment (sandbergja) [reply playbooks/repec.yml:10]: Good idea!  I added the datadog role.

Comment (sandbergja) [reply playbooks/repec.yml:10]: Bundler will handle pulling in the pul_assets repo.  The only thing to do is deploy the app, which doesn't seem worth a special `post_tasks` reminder to me.

**Review by maxkadel — APPROVED** [review]
Still approved! Thanks Jane!

**Pull request merged**